tests/test_large_set.py

```diff
@@ -0,0 +1,152 @@
+            and not (doc.name.endswith(SAFE_EXTENSION))
+            and re.match(FORMATS_REGEX, doc.name)
+        ]
+    )
+
+
+docs_10K = get_test_docs(min_size=0, max_size=10 * 2**10)
+docs_100K = get_test_docs(min_size=10 * 2**10, max_size=100 * 2**10)
+docs_10M = get_test_docs(min_size=100 * 2**10, max_size=10 * 2**20)
+docs_100M = get_test_docs(min_size=10 * 2**20, max_size=100 * 2**20)
+
+# Pytest parameter decorators
+for_each_10K_doc = pytest.mark.parametrize(
+    "doc", docs_10K, ids=[str(doc.name) for doc in docs_10K]
+)
+for_each_100K_doc = pytest.mark.parametrize(
+    "doc", docs_100K, ids=[str(doc.name) for doc in docs_100K]
+)
+for_each_10M_doc = pytest.mark.parametrize(
+    "doc", docs_10M, ids=[str(doc.name) for doc in docs_10M]
+)
+for_each_100M_doc = pytest.mark.parametrize(
+    "doc", docs_100M, ids=[str(doc.name) for doc in docs_100M]
+)
+
+
+@pytest.fixture
+def training(request: FixtureRequest) -> bool:
+    if request.config.getoption("--train"):
+        return True
+    else:
+        return False
+
+
+class TestLargeSet(TestCli):
+    def expected_container_output(self, input_file: Path) -> str:
+        # obtains the expected .log file
+        output_log_path = f"{input_file}.log"
+        with open(output_log_path, "r") as f:
+            return f.read()
+
+    def expected_success(self, input_file: Path) -> bool:
+        # obtains the expected result
+        expected_result_path = f"{input_file}.{CONTAINER_LOG_EXT}"
+        with open(expected_result_path, "r") as f:
+            last_line = f.readlines()[-1]  # result is in the last line
+            if "FAILURE" in last_line:
+                return False
+            elif "SUCCESS" in last_line:
+                return True
+            else:
+                raise ValueError(
+                    f"Container log file ({expected_result_path}) does not contain the result"
+                )
+
+    def run_doc_test(self, doc: Path, tmp_path: Path) -> None:
+        output_file_path = str(tmp_path / "output.pdf")
+        result = self.run_cli(
+            ["--output-filename", output_file_path, "--ocr-lang", "eng", str(doc)]
+        )
+        success = self.expected_success(doc)
+        if os.path.exists(output_file_path):
+            assert success, "Document was expected to fail but it didn't!"
+            result.assert_success()
+        else:
+            assert not success, "Document was expected to succeed but it didn't!"
+            result.assert_failure()
+
+    def train_doc_test(self, doc: Path, tmp_path: Path) -> None:
+        container_log_path = f"{doc}.{CONTAINER_LOG_EXT}"
+        if Path(container_log_path).exists():
+            os.remove(f"{doc}.{CONTAINER_LOG_EXT}")
+        output_file_path = str(tmp_path / "output.pdf")
+        self.run_cli(
+            ["--output-filename", output_file_path, "--ocr-lang", "eng", str(doc)],
+            env={"DZ_LOG_CONTAINER": "yes"},
+        )
```

Comment:
Some quick suggestions about the training stage:

1. When I first heard "train", due to some ML background, I always assumed that it would involve a computation in a subset of the documents, and then evaluating the results against the whole. We _do_ revisit the results, so you're right in this regard, but I think "record" would be more natural. Not a strong opinion though.
2. Even in training mode, should we perhaps do one of the tests in the normal mode. For instance, shouldn't we check that if the command succeeds, there is an output pdf as well?

Reply:
Also, just to be 100% sure on this. The training stage is used for two things:

1. Capture which docs fail and which succeed.
2. Capture the logs of the runs for future inspection. These logs are not actually used in a test though, right? Just the result part.

Reply:
It seems that the training phase does not exist in the latest iteration, which looks reasonable to me. Quick question before I resolve this conversion: I see the following in `TESTING.md`:

> The second kind of report is used to analyze the errors generated during the conversion. It can only be done after **test training**.

Do you mean after running the tests for the large document set?

Reply:
Removed in [e9882bf](https://github.com/freedomofpress/dangerzone/pull/386/commits/e9882bf2f6c683c574b445f3ddab5fea1a23a45a).